src/script/lua_api/l_object.cpp

```diff
@@ -1197,6 +1197,40 @@ int ObjectRef::l_get_fov(lua_State *L)
 	return 3;
 }
 
+// set_fog(self, distance)
+int ObjectRef::l_set_fog(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	ObjectRef *ref = checkObject<ObjectRef>(L, 1);
+	RemotePlayer *player = getplayer(ref);
+	if (player == nullptr)
+		return 0;
+
+	float distance = static_cast<f32>(luaL_checknumber(L, 2));
+	distance = rangelim(distance, 0.f, g_settings->getFloat("viewing_range"));
```

Comment:
Isn't this the server-side setting?

Reply:
`viewing_range` is a client-side setting.

Reply:
Yes, but you're using this here at the server-side.

Reply:
There are a hierarchical layers of Settings. This one is on the client.

Reply:
Ah, probably that if I call it from `l_object`, it'll try to get it from the server. Not sure. @sfan5 ?

Reply:
The only setting hierarchy I know is that for example world settings take preference over the global minetest.conf. (And idk of any client-related layer.)
But this hierarchy isn't relevant here. This code is executed on the server, which doesn't have access to the player's settings.